packages/cicero-core/src/contract.js

```diff
@@ -26,6 +31,138 @@ const TemplateInstance = require('./templateinstance.js');
  */
 class Contract extends TemplateInstance {
 
+    /**
+     * Create the Clause and link it to a Template.
+     * @param {Template} template  - the template for the clause
+     */
+     constructor(template) {
+        super(template);
+    }
+
+     /**
+     * Sign Instance
+     * @param {string} contractText - contract text extracted from contract markdown
+     * @param {object} signatureObject - contains signatures if existing parties who signed the contract. null if no one hasn't signed.
+     * @param {string} keyStorePath - path of the keystore to be used
+     * @param {string} keyStorePassword - password for the keystore file
+     * @return {object} object conatining array of all signatures
+     */
+      signInstance(contractText, signatureObject, keyStorePath, keyStorePassword) {
+        const ciceroMarkTransformer = new CiceroMarkTransformer();
+        const dom = ciceroMarkTransformer.fromMarkdownCicero( contractText, 'json' );
+        const resultText = this.formatCiceroMark(dom);
+        const hasher = crypto.createHash('sha256');
+        hasher.update(resultText);
+        const instanceHash = hasher.digest('hex');
+        
+        if(signatureObject !== null){
+            const contractHash = signatureObject.contractSignatures[0].contractHash;
+            if(instanceHash === contractHash){
+                const newSignatureObject = this.applySignature(instanceHash, keyStorePath, keyStorePassword);
+                const signatureArray = signatureObject.contractSignatures.concat(newSignatureObject);
+                const returnObject = {
+                    contractSignatures : signatureArray
+                }
+                return returnObject;
+            }else{
+                return 'Signature failed as the agreed contract was changed.'
+            }
+        }else{
```

Comment:
May need to update the linter for whitespace things 😃 I think `if (` and `} else {` would be preferred.